docs/assemblies/assembly-schema-validation-filter.adoc

```diff
@@ -0,0 +1,27 @@
+// file included in the following:
+//
+// assembly-built-in-filters.adoc
+
+[id='assembly-schema-validation-filter-{context}']
+= (Preview) Record Validation
+
+[role="_abstract"]
+The Record Validation filter validates records sent by a producer.   Only records that pass the validation are sent to
+the broker. This filter can be used to prevent _poison messages_ entering the Kafka system, that may otherwise lead to
+consumer failure.
+
+The filter currently supports two modes of operation:
+
+1. Schema validation validates the content of the record against a schema stored in an https://www.apicur.io/registry/[Apicurio Registry].
+2. Syntactically Correct JSON validation ensures the producer is producing messages that contain syntactically valid JSON.
```

Comment:
I wonder if the second mode should be just "Syntax Validation"? Then we have two similar sounding modes "Schema Validation mode" and "Syntax Validation mode". And maybe the "correct" is a bit redundant with the "validation" anyway?

Reply:
It only validates JSON, so it would need to be "JSON Syntax Validation".   Schema Validation is currently limited to JSON too but the intent is there for it to be extended to Avro and Protobuf.    It is very unlikely Syntax Validation will ever by generalised.

Reply:
pushed up a change.
@PaulRMellor how does that look?

Reply:
Fine Keith